### PR TITLE
Overlapping features using vector tiles

### PR DESCRIPTION
**Pull request by amishas157**
This PR adds a utility lib, which returns features in a tile given a tile identifier and a CF which finds overlapping features of an existing feature.

Next  action is to add incoming filters for this CF.

cc @bkowshik @geohacker @batpad

### CODE REVIEW

File: lib/feature-collection.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+'use strict';
```
Comment (geohacker) [inline lib/feature-collection.js]: @amishas157 can we rename this function to lib/getVectorTile or something to indicate that this is fetching a tile and returning features?

Comment (amishas157) [inline lib/feature-collection.js]: Yes. Sure

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+machine:
+  node:
+    version: 5.12.0
```
Comment (bkowshik) [inline circle.yml]: A couple of questions:
- What functionality needs node version `5.12.0`?
- What happens when consumers of this module use node version other than `5.12.0`. Ex: `4.x` or `6.x`?

File: comparators/feature_overlap.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+          });
+        });
+      });
+    }
+  } else if (featureCollections) {
+    featureCollections.forEach(function (featureCollection) {
+      featureCollection.features.forEach(function (feature) {
+        var intersection = intersect(incomingFeature, feature);
+        if (intersection) {
+          if (area(intersection) > 0) {
```
Comment (bkowshik) [inline comparators/feature_overlap.js]: Would it make it little easier to read if we move the area check along with the intersection check? Ex:

```js
if (intersection && (area(intersection) > 0)) {
    ...
}
```

File: comparators/feature_overlap.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            var id = feature.id.toString();
+            id = parseInt(id.substring(0, id.length - 1));
```
Comment (bkowshik) [inline comparators/feature_overlap.js]: Don't understand why this is not a straightforward `parseInt`. An example of the format with a small description as a comment should be super helpful to have here.

File: comparators/feature_overlap.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            if (relationMembers.indexOf(id) === -1) {
```
Comment (bkowshik) [inline comparators/feature_overlap.js]: I am 🤔 this `if` condition will always be `true` as the `relationMembers` array is populated only in the `if` block and not in the current `else` block`. What do you think?

File: tests/fixtures/feature_overlap.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            ]
+          ]
+        }
+      },
+      "oldVersion": {},
+      "expectedResult": {
+        "result:feature_overlap": 24
+      }
+    }
+  ]
```
Comment (bkowshik) [inline tests/fixtures/feature_overlap.json]: What does a failing test case look like, i.e: When there is no overlap.

File: circle.yml (hunk 2 above)
Comment (amishas157) [inline circle.yml]: Hey @bkowshik 👋 Sorry for replying late. So this is kind of circleci + node + ubuntu version deal. 
So as per  https://circleci.com/gh/mapbox/osm-compare/edit#build-environment, we are using ubuntu 12.04
And it supports discrete versions for node which are listed here. https://circleci.com/docs/1.0/build-image-precise/#nodejs
And for now we are using the latest version supported by 12.04 ubuntu i.e. `5.5.0`

>What functionality needs node version 5.12.0?

  When we don't have any circle.yml configuration set, we get issue with [vt2geojson](https://github.com/mapbox/vt2geojson)
`TypeError: zlib has no method 'gunzipSync'` 

> What happens when consumers of this module use node version other than 5.12.0. Ex: 4.x or 6.x?

The error which is coming with vt2geojson is resolved with node version `v0.11.12`. [Ref](https://nodejs.org/dist/latest-v7.x/docs/api/zlib.html#zlib_zlib_gunzipsync_buf_options) . By default circleci uses node version `0.10.33` which is mentioned [here](https://circleci.com/docs/1.0/build-image-precise/#nodejs).

File: tests/fixtures/feature_overlap.json (hunk 6 above)
Comment (amishas157) [inline tests/fixtures/feature_overlap.json]: Yes, that's sounds like a good idea, we should have test for falsey results as well.

File: circle.yml (hunk 2 above)
Comment (batpad) [inline circle.yml]: sorry for being late to this party, but @amishas157 @bkowshik - is there a reason we are using ubuntu 12.04? We should only need to support ubuntu 14.04 at the earliest. 12.04 is now over 5 years old, and I believe coming out of its LTS support period.

**Pull request merged**